simulator/source/Combat_simulator.cpp

```diff
@@ -1547,6 +1570,7 @@ void Combat_simulator::simulate(const Character& character, int init_iteration,
                     }
                     else
                     {
+                        update_swing_timers(weapons[0], is_dual_wield ? weapons[1] : weapons[0], oldHaste, special_stats.haste);
```

Comment:
this is the actual tight loop culprit: if you get here while slam isn't casting, the swing timers never get updated, and this repeats ad infinitum

note: this essentially prefers slam over all other abilities, typically waiting for the next main hand swing